qemu/tests/qemu_guest_agent.py

```diff
@@ -126,18 +125,24 @@ def _check_ga_pkg(self, session, cmd_check_pkg):
                               logging.info)
         s, o = session.cmd_status_output(cmd_check_pkg)
         if s == 0 and self.params.get("os_variant", "") == 'rhel8':
-            # qemu-guest-agent-2.12.0-88.module+el8.1.0+4233+bc44be3f.x86_64
-            error_context.context("Check if the installed pkg is the latest"
+            error_context.context("Check if the installed pkg is the specific"
                                   " one for rhel8 guest.", logging.info)
             version_list = []
-            build_latest = re.sub(r'/', '-', self.qga_pkg_latest_url)
-            for pkg in [o, build_latest]:
-                pattern = r"guest-agent-(\d+.\d+.\d+-\d+).module"
-                qga_v = re.findall(pattern, pkg, re.I)[0]
-                version_list.append(qga_v)
-            logging.info("The installed and the latest pkg version is"
-                         " %s", version_list)
-            if version_list[1] != version_list[0]:
+            full_qga_rpm_info = session.cmd("readlink %s" % self.qga_pkg_path)
+            build_specific = re.sub(r'/', '-', full_qga_rpm_info)
+            if full_qga_rpm_info:
+                for pkg in [o, build_specific]:
+                    pattern = r"guest-agent-(\d+.\d+.\d+-\d+).module"
+                    qga_v = re.findall(pattern, pkg, re.I)
+                    if qga_v:
+                        version_list.append(qga_v[0])
+                        logging.info("The installed and the specific pkg "
```

Comment:
It should output the the version info after the end of the loop.

Reply:
Despite the dict and list issue, I think it should at least be moved out from the `for` loop. Now the info will be print twice and the first round is not correct.

Reply:
Yes~   This is not controversial ~   @6-dehan

Reply:
> Despite the dict and list issue, I think it should at least be moved out from the `for` loop. Now the info will be print twice and the first round is not correct.

@vivianQizhu @yanan-fu the logging.info has change out of the `for` loog, here are result of new code. I checked the log that the qga_v info print once as expected.
(01/50) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (937.98 s)
 (02/50) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_install.q35: PASS (95.89 s)